src/cut_variation.py

```diff
@@ -0,0 +1,419 @@
+                        hCorrYieldPrompt.GetBinLowEdge(nPtBins) + hCorrYieldPrompt.GetBinWidth(nPtBins),
+                        hCorrYieldPrompt.GetMaximum() * 1.2, ';#it{p}_{T} (GeV/#it{c});corrected yield')
+    cCorrYield.SetLogy()
+    hCorrYieldPrompt.Draw('same')
+    hCorrYieldFD.Draw('same')
+    legEff.Draw()
+
+    if systOpt:
+        outDir = inputPath.replace('eff', 'syst_frac') + f'/{systOpt}'
+    else:
+        outDir = inputPath.replace('eff', 'cutVar')
+    os.makedirs(outDir, exist_ok=True)
+    outFileName = f'{outDir}/cutVar.root'
+    outFile = TFile(outFileName, 'recreate')
+    cCorrYield.Write()
+    hCorrYieldPrompt.Write()
+    hCorrYieldFD.Write()
+    for covElem in product(range(2), range(2)):
+        hCovCorrYields[covElem[0]][covElem[1]].Write()
+    for iPt, (ptmin, ptmax) in enumerate(zip(ptmins, ptmaxs)):
+        logger(f'Saving cut variation results for pt {ptmin:.1f}-{ptmax:.1f} GeV/c')
+        outFile.mkdir(f"pt{ptmin:.1f}_{ptmax:.1f}")
+        outFile.cd(f"pt{ptmin:.1f}_{ptmax:.1f}")
+        cFinalResPt[iPt].Write()
+        hRawYieldsVsCut[iPt].Write()
+        hRawYieldPromptVsCut[iPt].Write()
+        hRawYieldFDVsCut[iPt].Write()
+        hRawYieldsVsCutReSum[iPt].Write()
+        hEffPromptVsCut[iPt].Write()
+        hEffFDVsCut[iPt].Write()
+        hPromptFracVsCut[iPt].Write()
+        hFDFracVsCut[iPt].Write()
+        hCorrMatrixCutSets[iPt].Write()
+    outFile.Close()
+
+    for iPt in range(len(ptmins)):
+        if iPt == 0:
+            cFinalResPt[iPt].SaveAs(f'{outDir}/FinalResPt.pdf[')
+        cFinalResPt[iPt].SaveAs(f'{outDir}/FinalResPt.pdf')
+        if iPt == hRawYields[0].GetNbinsX() - 1:
+            cFinalResPt[iPt].SaveAs(f'{outDir}/FinalResPt.pdf]')
+        cFinalResPt[iPt].SaveAs(f'{outDir}/FinalResPt_pt{ptmins[iPt]}_{ptmaxs[iPt]}.png')
+    
+
+def compute_frac_cut_var(config_flow, inputPathEff, inputPathRy, correlated=False, batch=False):
+
+    gROOT.SetBatch(batch)
+    gStyle.SetPaintTextFormat("4.2f")
+
+    # load configuration
+    ptmins, ptmaxs = [], []
+    with open(config_flow, 'r') as ymlCfgFile:
+        config = yaml.safe_load(ymlCfgFile)
+        ptmins = config['ptbins'][:-1]
+        ptmaxs = config['ptbins'][1:]
+
+    if os.path.exists(inputPathEff):
+        effFiles = [f'{os.path.join(inputPathEff, file)}'
+                    for file in os.listdir(inputPathEff) if file.startswith('eff_') and file.endswith('.root')]
+    else:
+        logger(f'No eff folder found in {inputPathEff}', level='ERROR')
+        raise ValueError(f'No eff folder found in {inputPathEff}')
+
+    if os.path.exists(inputPathRy):
+        rawYieldFiles = [f'{os.path.join(inputPathRy, file)}'
+                         for file in os.listdir(inputPathRy) if file.startswith('raw_yields_') and file.endswith('.root')]
+    else:
+        logger(f'No vn folder found in {inputPathRy}', level='ERROR')
+        raise ValueError(f'No vn folder found in {inputPathRy}')
```

Comment:
```suggestion
    if not os.path.exists(inputPathEff):
            logger(f'No eff folder found in {inputPathEff}', level='ERROR')
            raise ValueError(f'No eff folder found in {inputPathEff}')
    if not os.path.exists(inputPathRy)::
            logger(f'No vn folder found in {inputPathRy}', level='ERROR')
            raise ValueError(f'No vn folder found in {inputPathRy}')
        
    effFiles = [f'{os.path.join(inputPathEff, file)}'
                    for file in os.listdir(inputPathEff) if file.startswith('eff_') and file.endswith('.root')]
    rawYieldFiles = [f'{os.path.join(inputPathRy, file)}'
                         for file in os.listdir(inputPathRy) if file.startswith('raw_yields_') and file.endswith('.root')]
```

I suggest this version as it will fail immediately and we diminish the nested loops

Reply:
True, and I will move all of them into a separate script in `utils`.